akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -80,9 +82,10 @@ object One2OneBidiFlow {
             }
           } else throw new UnexpectedOutputException(element)
         }
-        override def onUpstreamFinish(): Unit =
-          if (pending == 0 && !innerFlowCancelled) complete(outOut)
+        override def onUpstreamFinish(): Unit = {
+          if (pending == 0 && isClosed(inIn) && !innerFlowCancelled) complete(outOut)
```

Comment:
I doubt the purpose of this whole stage given that there is no way to handle cancellation such that it is both intuitive and correct. What is the point?